client/src/lib/attempt-result.js

```diff
@@ -197,6 +197,54 @@ export function formatAttemptResult(attemptResult, eventId) {
   return centisecondsToClockFormat(attemptResult);
 }
 
+/**
+ * Calcualte BPA (Best Possible Average) for the given attempts.
+ * Returns a human-friendly string.
+ * @example
+ * calculateBpa([6111, 6000, 5999, 6000], '333'); // => '1:00.00'
+ * calculateBpa([6111, -1, -1, 6000], '333'); // => 'DNF'
+ * calculateBpa([6111, -1, 6000, 5999], '333'); // => '1:00.36'
+ */
+export function calculateBpa(times, eventId) {
```

Comment:
Nitpick: it's generally a good idea to avoid acronyms unless they are very well establish. So I'd do this: `bestPossibleAverage`, `worstPossibleAverage`, `incompleteMean`.

Also note that the average needs to be rounded accordingly, and for FMC we want to multiply the attempts by 100 to represent 2 decimal places in the integer. See the implementation of `average`, we can closely match the rules from there.